# Propellant Calculation for Satellite Operation

Calculate the propellant required for an operation for a satellite with the drymass of 300 kg from the altitude of 435 km to 500 km. The specific impulseof the thruster is 200 seconds.

In [5]:

import math

# Constants
mu = 3.986e14  # Earth's gravitational parameter (m^3/s^2)
RE = 6371e3  # Earth's radius (m)
g0 = 9.81  # Standard gravitational acceleration (m/s^2)
Isp = 200  # Specific impulse (s)
dry_mass = 300  # Dry mass of the satellite (kg)

# Altitudes
h1 = 435e3  # Initial altitude (m)
h2 = 500e3  # Final altitude (m)

# Orbital velocities
r1 = RE + h1  # Initial orbital radius (m) = 6806 (km)
r2 = RE + h2  # Final orbital radius (m) = 6871 (km)
v1 = math.sqrt(mu / r1)  # Initial orbital velocity (m/s)
v2 = math.sqrt(mu / r2)  # Final orbital velocity (m/s)
delta_v = abs(v2 - v1)  # Change in velocity (m/s)

# Using Tsiolkovsky rocket equation
mass_ratio = math.exp(delta_v / (Isp * g0))
initial_mass = dry_mass * mass_ratio  # Total initial mass (kg)
propellant_mass = initial_mass - dry_mass  # Propellant mass (kg)

print(f'propellant mass = {propellant_mass} kg')


propellant mass = 5.599652845460014 kg


## Results

The change in velocity (Δv) is approximately **36.28 m/s**, and the required propellant mass is approximately **5.60 kg**.